src/apps/vpn/connectionmanager.h

```diff
@@ -0,0 +1,244 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef CONNECTIONMANAGER_H
+#define CONNECTIONMANAGER_H
+
+#include <QDateTime>
+#include <QList>
+#include <QObject>
+#include <QTimer>
+
+#include "interfaceconfig.h"
+#include "ipaddress.h"
+#include "loghandler.h"
+#include "models/server.h"
+#include "models/serverdata.h"
+#include "pinghelper.h"
+
+class ControllerImpl;
+class MozillaVPN;
+
+class ConnectionManager : public QObject, public LogSerializer {
+  Q_OBJECT
+  Q_DISABLE_COPY_MOVE(ConnectionManager)
+
+ public:
+  //  enum ConnectionState {
+  //    ConnectionStateInternetProbe,
+  //    ConnectionStateServerProbe,
+  //    ConnectionStateFirewall,
+  //    ConnectionStateCaptivePortal,
+  //    ConnectionStateCheckSubscription,
+  //    ConnectionStateUnstable,
+  //    ConnectionStateIdle,       // Used when there are no active probes
+  //    ongoing
+  //  };
+  //  Q_ENUM(ConnectionState)
+
+  // These are temporarily copied from the controller state to mirror them, but
```

Comment:
Do we use TODO flags along with tracking ticket numbers in the source code to track pending work? Or is this OK as is because we are sure that this change will be made soon?

Reply:
I'm not aware of a standard process for TODOs but I'd be happy to add a TODO flag for this. Let me know if the one I added looks okay or if there's a better way to do this

Reply:
The code base has some instances of the following pattern which references a ticket that tracks remaining work:
// TODO: Implement this once we update to Qt6.3 (VPN-3511)